.github/workflows/tests.yml

```diff
@@ -46,33 +33,42 @@ jobs:
           - "3.11"
           - "pypy-3.8"
           - "pypy-3.9"
+        os:
+          - name: Linux
+            runs-on: [ubuntu-latest]
+          - name: macOS
+            runs-on: [macos-latest]
+          - name: Windows
+            runs-on: [windows-latest]
+        include:
+          - tox:
+              name: Flake8
+              environment: flake8
+            python: "3.11"
+            os:
+              name: Linux
+              runs-on: [ubuntu-latest]
+          - tox:
+              name: Docs
+              environment: docs
+            python: "3.11"
+            os:
+              name: Linux
+              runs-on: [ubuntu-latest]
+
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-        with:
-          ref: ${{ github.event.inputs.branch }}
-      - name: Use Python ${{ matrix.python }}
+
+      - name: Set up Python ${{ matrix.python }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
-      - name: Install test dependencies
-        run: python -m pip install tox
-      - name: Test
-        run: python -m tox -e py
 
-  docs:
-    name: Test documentation builds
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-        with:
-          ref: ${{ github.event.inputs.branch }}
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.11"
       - name: Install test dependencies
-        run: python -m pip install tox
-      - name: Test
-        run: python -m tox -e docs
+        run: |
```

Comment:
Why not the command directly?

Reply:
To be sure I am understanding correctly, you mean `run: python -m tox ...`?  Some were one way and some the other.  I personally always use `run: |` so that when I remove a line and end up at one or add a line from having just one, I do not have to flip-flop on the use of `|` or not.  But sure, definitely a matter of preference here.

Reply:
Oh OK. It's fine for me as is.